tests/phpunit/integration/Modules/Analytics_4Test.php

```diff
@@ -206,13 +237,848 @@ public function test_get_datapoints() {
+		);
+
+		$http_client = $this->create_fake_http_client( $property_id );
+		$this->analytics->get_client()->setHttpClient( $http_client );
+		$this->analytics->register();
+
+		$this->set_shareable_metrics( 'sessions', 'totalUsers' );
+
+		$this->enable_shared_credentials();
+
+		$data = $this->analytics->get_data(
+			'report',
+			array(
+				// Note, metrics is a required parameter.
+				'metrics' => array(
+					array( 'name' => 'sessions' ),
+					array( 'name' => 'totalUsers' ),
+					array( 'name' => 'invalidMetric' ),
+					array( 'name' => 'anotherInvalidMetric' ),
+				),
+			)
+		);
+
+		$this->assertWPErrorWithMessage( 'Unsupported metrics requested: invalidMetric, anotherInvalidMetric', $data );
+	}
+
+	public function test_report__dimension_validation() {
+		$this->enable_feature( 'dashboardSharing' );
+
+		$this->context        = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$options              = new Options( $this->context );
+		$this->user           = $this->factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		$user_options         = new User_Options( $this->context, $this->user->ID );
+		$this->authentication = new Authentication( $this->context, $options, $user_options );
+		$this->analytics      = new Analytics_4( $this->context, $options, $user_options, $this->authentication );
+
+		// Re-register Permissions after enabling the dashboardSharing feature to include dashboard sharing capabilities.
+		// TODO: Remove this when `dashboardSharing` feature flag is removed.
+		$modules     = new Modules( $this->context, null, $user_options, $this->authentication );
+		$permissions = new Permissions( $this->context, $this->authentication, $modules, $user_options, new Dismissed_Items( $user_options ) );
+		$permissions->register();
+
+		$property_id = '123456789';
+
+		$this->analytics->get_settings()->merge(
+			array(
+				'propertyID' => $property_id,
+			)
+		);
+
+		$http_client = $this->create_fake_http_client( $property_id );
+		$this->analytics->get_client()->setHttpClient( $http_client );
+		$this->analytics->register();
+
+		$this->set_shareable_metrics( 'sessions' );
+		$this->set_shareable_dimensions( 'date', 'pageTitle' );
+
+		$this->enable_shared_credentials();
+
+		$data = $this->analytics->get_data(
+			'report',
+			array(
+				// Note, metrics is a required parameter.
+				'metrics'    => array(
+					array( 'name' => 'sessions' ),
+				),
+				'dimensions' => array( 'date', 'pageTitle', 'invalidDimension', 'anotherInvalidDimension' ),
+			)
+		);
+
+		$this->assertWPErrorWithMessage( 'Unsupported dimensions requested: invalidDimension, anotherInvalidDimension', $data );
+	}
+
+	/**
+	 * Returns a date string for the given number of days ago.
+	 *
+	 * @param int $days_ago The number of days ago.
+	 * @return string The date string, formatted as YYYY-MM-DD.
+	 */
+	protected function days_ago_date_string( $days_ago ) {
+		return gmdate( 'Y-m-d', strtotime( $days_ago . ' days ago' ) );
+	}
```

Comment:
I've ended up leaving this in, due to the new default date range test case mentioned above.

I started looking for solutions for mocking the date in PHPUnit but didn't want to go too far down a rabbit hole.

Given the low probability of hitting the edge case you mentioned on a given test run, I'd suggest leaving it in for now and following up with a separate issue to provide a date mocking solution and rewrite the test, as this would also provide useful functionality for other tests. What do you think?

Reply:
Sure, that sounds good to me. It's very unlikely that the test would fail for this reason, but I agree we should preserve the default date range 👍

Reply:
Ok, cool. I've created the issue #6407 to follow up on mocking the date in PHPUnit tests.